x/btcstaking/keeper/finality_providers.go

```diff
@@ -76,6 +84,12 @@ func (k Keeper) setFinalityProvider(ctx context.Context, fp *types.FinalityProvi
 	store.Set(fp.BtcPk.MustMarshal(), fpBytes)
 }
 
+func (k Keeper) bsnIndexFinalityProvider(ctx context.Context, fp *types.FinalityProvider) {
+	indexStore := k.finalityProviderBsnIndexStore(ctx)
+	bsnKey := types.BuildBsnIndexKey(fp.BsnId, fp.BtcPk)
+	indexStore.Set(bsnKey, []byte{}) // Empty value, just for existence
+}
```

Comment:
To achieve the goal "given BSN ID, output list of BTC PKs of FPs securing this BSN", I think we can use the `KeySet` type provided since Cosmos SDK v0.50. See an example [here](https://github.com/babylonlabs-io/babylon/blob/d253be02ef2fb8443f01760f6714064003d6b469/x/incentive/keeper/keeper.go#L24-L26). This will be much cleaner compared to the old manner prior to v0.50

Reply:
I'll add it as a TODO for after the PR is merged